### ISSUE TITLE
[live_grep] no highlights in results window for a regex query

### ISSUE THREAD

**Issue by yanzhang0219** [nvim-telescope/telescope.nvim#1708]
### Description

In `live_grep`, when I type a regex string as query after the prompt, no highlight for the matching in the results window.

### Neovim version

```markdown
NVIM v0.7.0-dev+797-gb218d02c4
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

macOS 12.1 (the latest)

### checkhealth telescope

```markdown
telescope: require("telescope.health").check()
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - OK: nvim-treesitter installed.

## Checking external dependencies
  - OK: rg: found ripgrep 13.0.0
  - OK: fd: found fd 8.3.1

## ===== Installed extensions =====

## Telescope Extension: `fzf`
  - INFO: No healthcheck provided
```

### Steps to reproduce

1. launch `live_grep`
2. type a string not containing regex, in the results window all the matching will be highlighted
3. type a regex string, in the result window no highlight for all the matching

### Expected behavior

All the matching should be highlighted as well when the query is regex.

### Actual behavior

<img width="1147" alt="Screen Shot 2022-01-21 at 13 14 05" src="https://user-images.githubusercontent.com/11582667/150601024-7b593caa-cbda-4201-9f75-06e43d6be013.png">

However, if I type a normal string as a query, all the matching will be highlighted

<img width="1147" alt="Screen Shot 2022-01-21 at 13 15 28" src="https://user-images.githubusercontent.com/11582667/150601129-e6a5d396-f196-47eb-afc8-100eb9d8e3f6.png">

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('fzf')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by fdschmidt93** [comment]
Highlighting in the results window is actually performed by a fzy sorter.

Essentially, somebody would have to pick up https://github.com/nvim-telescope/telescope.nvim/pull/1448

**Comment by yanzhang0219** [comment]
Thanks. Btw, I use fzf as the sorter. 
Should I try telescope-rg.nvim?

**Comment by fdschmidt93** [comment]
Ah, I was too brief. The fzy sorter for highlighting pretty much only gets the offsets of matched prompts for highlighting but does not perform any sorting per se.

`live_grep` therefore doesn't practically speaking have any sorter. `grep_string` passes the string (also possibly an empty string) onto `rg` and then filters that one using your sorter.

> Should I try telescope-rg.nvim?

I think the authors originally planned on picking up the PR but never came around. That's why I mentioned somebody has to pick it up :) in the long run, in open source people ideally implement the features they want to see implemented (or generally have fun implementing stuff, of course).

**Comment by yanzhang0219** [comment]
Thank you @fdschmidt93  so much for your clear explanation. 

On more question (or issue) here still for `live_grep` highlight. 
When I type a query string, say `main`, only the files containing `main xxx` will be listed and `main` should be highlighted. However, if a file has name like `main.go` and its content has `main xxx`, highlight will be misplaced, i.e., `main` in the filename will be highlighted instead of the actual content. 

<img width="1147" alt="Screen Shot 2022-01-21 at 15 45 34" src="https://user-images.githubusercontent.com/11582667/150614298-9e96e98c-c8d0-4fc2-96fa-b1fef4f38112.png">

**Comment by fdschmidt93** [comment]
It's pretty much the same issue. If we got our highlighting from `rg` directly that issue would be fixed. I'd say you're happy to open another issue for that, though in practice would have the same solution I guess.

**Comment by yanzhang0219** [comment]
Okay, I got it. Thanks.

**Comment by mrbeardad** [comment]
Is there any progress? 👀

**Comment by Conni2461** [comment]
Yes there is progress here. One of our team members made a wip pr  https://github.com/nvim-telescope/telescope.nvim/pull/2536 

I hope we can finish this one soon
